rmf_gazebo_plugins/src/readonly.cpp

```diff
@@ -0,0 +1,466 @@
+  RCLCPP_INFO(logger(), "Setting level name to: " + _level_name);
+
+  if (sdf->HasElement("graph_index"))
+    _nav_graph_index = sdf->Get<std::size_t>("graph_index");
+  RCLCPP_INFO(logger(), "Setting nav graph index: " + _nav_graph_index);
+
+  if (sdf->HasElement("spawn_waypoint"))
+    _start_wp_name = sdf->Get<std::string>("spawn_waypoint");
+  RCLCPP_INFO(logger(), "Setting start wp name: " + _start_wp_name);
+
+  if (sdf->HasElement("look_ahead"))
+    _lookahead = sdf->Get<std::size_t>("look_ahead");
+  _lookahead = _lookahead < 1 ? 1 : _lookahead;
+  _next_wp.resize(_lookahead);
+  _path.resize(_lookahead);
+  RCLCPP_INFO(logger(), "Setting lookahead: " + std::to_string(_lookahead));
+
+  if (sdf->HasElement("nominal_drive_speed"))
+    _v_n = sdf->Get<double>("nominal_drive_speed");
+  RCLCPP_INFO(logger(), "Setting nominal drive speed to: " + std::to_string(_v_n));
+
+  if (sdf->HasElement("nominal_drive_acceleration"))
+    _a_n = sdf->Get<double>("nominal_drive_acceleration");
+  RCLCPP_INFO(logger(), "Setting nominal drive acceleration to: " + std::to_string(_a_n));
+
+  if (sdf->HasElement("nominal_turn_speed"))
+    _w_n = sdf->Get<double>("nominal_turn_speed");
+  RCLCPP_INFO(logger(), "Setting nominal turn speed to:" + std::to_string(_w_n));
+
+  if (sdf->HasElement("nominal_turn_acceleration"))
+    _alpha_n = sdf->Get<double>("nominal_turn_acceleration");
+  RCLCPP_INFO(logger(), "Setting nominal turn acceleration to:" + std::to_string(_alpha_n));
+
+  RCLCPP_INFO(logger(), "hello i am " + model->GetName());
+
+  _update_connection = gazebo::event::Events::ConnectWorldUpdateBegin(
+      std::bind(&ReadonlyPlugin::OnUpdate, this));
+
+  robot_state_pub = _ros_node->create_publisher<rmf_fleet_msgs::msg::RobotState>(
+      "/robot_state", 10);
+
+  // Subscription to /map
+  auto qos_profile = rclcpp::QoS(10);
+  qos_profile.transient_local();
+  _building_map_sub = _ros_node->create_subscription<BuildingMap>(
+        "/map",
+        qos_profile,
+        std::bind(&ReadonlyPlugin::map_cb, this, std::placeholders::_1));
+
+  _current_task_id = "demo";
+  if (model->GetName().c_str())
+  _name = _model->GetName().c_str();
+  RCLCPP_INFO(logger(), "Setting name to: " + _name);
+
+}
+
+void ReadonlyPlugin::map_cb(const BuildingMap::SharedPtr msg)
+{
+  if (msg->levels.empty())
+  {
+    RCLCPP_ERROR(logger(), "Received empty building map");
+    return;
+  }
+
+  _found_map = true;
+  _map = *msg;
+  RCLCPP_INFO(logger(), "Received building map with %d levels", msg->levels.size());
+  _found_level = false;
+  _found_graph = false;
+  for (const auto& level: msg->levels)
+  {
+    RCLCPP_INFO(logger(), "Level name: [%s]", level.name.c_str());
+    if (level.name.c_str() == _level_name)
+    {
+      _level = level;
+      _found_level = true;
+      RCLCPP_INFO(logger(), "Found level [%s] with %d nav_graphs",
+          level.name.c_str(), level.nav_graphs.size());
+
+      if (_nav_graph_index < level.nav_graphs.size())
```

Comment:
this condition might not be enough to determine whether the graph is valid, especially for cases like the `airport_terminal` map where the graphs exist on index `1` and `2` while `0` is empty. Do you think it might make more sense to check if the graph has at least a small number of waypoints?

Reply:
Although the `graph_idx` 0 was skipped in the traffic editor model, the BuildingMap msg published still contains only two graphs: indexed at 0 and  1 resp. I does not publish an empty graph representative of the skipped one. To follow convention, perhaps we should go back to the traffic editor model and correct the starting index of the graphs?

Reply:
interesting, I didn't know `building_map` will publish the graphs in that way. 
sure thing, we can have a fix branch that does that once this gets merged. This line of code is valid then